src/mongocxx/options/create_collection.hpp

```diff
@@ -0,0 +1,119 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongocxx/config/prelude.hpp>
+
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+#include <mongocxx/stdx.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+namespace options {
+
+///
+/// Class representing the optional arguments to a MongoDB createCollection command
+///
+class MONGOCXX_API create_collection {
+   public:
+    ///
+    /// To create a capped collection, specify true.
+    ///
+    /// @note If you specify true, you must also set a maximum size using the size() method.
```

Comment:
After a brief drivers survey, I'm not going to add validation for this now.  Driver authors do a range of things, and since there's no spec nobody really knows what the right behavior is.  The C driver does do some validation but not the validation described in our docs.